frontend/src/dashboard.js

```diff
@@ -46,623 +44,528 @@ import {
   ResultSummaryWidget
 } from './widgets';
 import { IbutsuContext } from './services/context.js';
+import { useNavigate, useParams } from 'react-router-dom';
 
 
-export class Dashboard extends React.Component {
-  static contextType = IbutsuContext;
-  static propTypes = {
-    eventEmitter: PropTypes.object,
-    navigate: PropTypes.func,
-    params: PropTypes.object,
-  }
+function Dashboard() {
+  const context = useContext(IbutsuContext);
```

Comment:
I'd suggest just destructuring what you need here directly (#nit)

Reply:
absolutely makes sense, I started with this but don't really use it "directly".

Reply:
Looks like I can apply the same pattern to my use of `params` hook as well.